api/src/pcapi/core/educational/api/dms.py

```diff
@@ -0,0 +1,165 @@
+from datetime import timedelta
+import logging
+import typing
+
+from pcapi import settings
+from pcapi.connectors.dms import api as dms_api
+from pcapi.connectors.dms import models as dms_models
+from pcapi.core.educational import models as educational_models
+from pcapi.core.offerers import models as offerers_models
+from pcapi.models import db
+
+
+logger = logging.getLogger(__name__)
+
+
+def import_dms_application(procedure_number: int = 0) -> None:
+    """import dms applications for eac status"""
+    if procedure_number:
+        procedures = [procedure_number]
+    else:
+        procedures = [
+            settings.DMS_EAC_PROCEDURE_INDEPENDANTS_CANDIDATE_ID,
+            settings.DMS_EAC_PROCEDURE_STRUCTURE_CANDIDATE_ID,
+            settings.DMS_EAC_PROCEDURE_MENJS_CANDIDATE_ID,
+        ]
+
+    for procedure in procedures:
+        previous_import = _get_previous_import(procedure)
+        if previous_import and previous_import.isProcessing:
+            logger.info(
+                "[DMS] Procedure %s is already being processed.",
+                procedure,
+                extra={
+                    "procedure_number": procedure,
+                },
+            )
+            continue
+        since = previous_import.latestImportDatetime if previous_import else None
+
+        update_dms_applications_for_procedure(
+            procedure_number=procedure,
+            since=since,
+        )
+
+
+def update_dms_applications_for_procedure(procedure_number: int, since: datetime | None) -> None:
+    logger.info(
+        "[DMS] Starting processing procedure %s.",
+        procedure_number,
+        extra={
+            "procedure_number": procedure_number,
+        },
+    )
+
+    current_import = dms_models.LatestDmsImport(
+        procedureId=procedure_number,
+        latestImportDatetime=datetime.utcnow(),
+        isProcessing=True,
+        processedApplications=[],
+    )
+    db.session.add(current_import)
+    db.session.commit()
+
+    dms_client = dms_api.DMSGraphQLClient()
+    applications_id = []
+
+    for node in dms_client.get_eac_nodes_siret_states(procedure_number=procedure_number, since=since):
+        try:
+            _get_or_create_application(procedure_number=procedure_number, node=node)
+        except Exception as exp:  # pylint: disable=broad-exception-caught
+            logger.error(
+                "[DMS] Application parsing failed with error %s",
+                str(exp),
+                extra={
+                    "application_number": node.get("number"),
+                    "application_scalar_id": node.get("id"),
+                    "procedure_number": procedure_number,
+                },
+            )
+        else:
+            applications_id.append(node["number"])
+
+    current_import.processedApplications = applications_id
+    current_import.isProcessing = False
+    db.session.commit()
+
+    logger.info(
+        "[DMS] Finishing processing procedure %s.",
```

Comment:
(micro-détail) On n'est plus "_en train_ de terminer", on **a** terminé. ;)

```suggestion
        "[DMS] Finished processing EAC procedure %s.",
```